File: moderate/remove_chars.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my ( $l, $r ) = split( /, /, $line );
    $l =~ s/[$r]//g;
    print "$l\n";
}
close(INFILE);

File: moderate/remove_chars.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

func removeChars(p, q string) string {
	mapf := func(r rune) rune {
		if strings.Contains(q, string(r)) {
			return -1
		}
		return r
	}
	return strings.Map(mapf, p)
}

func main() {
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		s := strings.Split(scanner.Text(), ", ")
		fmt.Println(removeChars(s[0], s[1]))
	}
}
